pkg/api/config/config.go

```diff
@@ -27,15 +27,36 @@ type StorageConfig struct {
 	Dedupe                      bool
 	RemoteCache                 bool
 	GC                          bool
-	Commit                      bool
 	GCDelay                     time.Duration
 	GCInterval                  time.Duration
-	GCReferrers                 bool
 	UntaggedImageRetentionDelay time.Duration
+	Retention                   ImageRetention
+	Commit                      bool
 	StorageDriver               map[string]interface{} `mapstructure:",omitempty"`
 	CacheDriver                 map[string]interface{} `mapstructure:",omitempty"`
 }
 
+type ImageRetention struct {
+	DryRun   bool
+	Policies []GCPolicy
+}
+
+type GCPolicy struct {
+	RepoNames       []string
+	DeleteReferrers bool
+	DeleteUntagged  bool
+	TagsRetention   []TagsRetentionPolicy
+}
+
+type TagsRetentionPolicy struct {
+	Names                   []string
+	RetainAlways            bool // default is true
+	PulledWithinLastNrDays  *int
```

Comment:
why "*int"

Reply:
so I can check for nil value, but I guess I can also skip 0 values.